ako-infra/avirest/handle_netinfo.go

```diff
@@ -0,0 +1,250 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package avirest
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	avicache "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/cache"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/rest"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/third_party/github.com/vmware/alb-sdk/go/clients"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/third_party/github.com/vmware/alb-sdk/go/session"
+
+	"github.com/vmware/alb-sdk/go/models"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var CloudCache *models.Cloud
+
+// SyncLSLR fetches all networkinfo CR objects, compares them with the data network configured in the cloud,
+// and updates the cloud if any LS-LR data is missing.
+func SyncLSLR() {
+	lslrmap := lib.GetNetinfoCRData()
+	utils.AviLog.Infof("Got data LS LR Map: %v, from NetworkInfo CR", lslrmap)
+
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("Failed to get Cloud data from cache")
+		return
+	}
+	matched := matchSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lslrmap)
+	if !matched {
+		cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs = constructLsLrInCloud(lslrmap)
+		path := "/api/cloud/" + *cloudModel.UUID
+		restOp := utils.RestOp{ObjName: utils.CloudName, Path: path, Method: utils.RestPut, Obj: &cloudModel,
+			Tenant: "admin", Model: "cloud", Version: utils.CtrlVersion}
+		executeRestOp("fullsync", client, &restOp)
+	} else {
+		utils.AviLog.Infof("LS LR update not required in cloud: %s", utils.CloudName)
+	}
+}
+
+func AddSegment(obj interface{}) {
+	objKey := "Netinfo" + utils.ObjKey(obj)
+	utils.AviLog.Debugf("key: %s, Network Info DELETE Event", objKey)
+	crd := obj.(*unstructured.Unstructured)
+
+	specJSON, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "topology")
+	if err != nil || !found {
+		utils.AviLog.Warnf("NetworkInfo topology not found: %+v", err)
+		return
+	}
+	lr := specJSON["gatewayPath"]
+	ls := specJSON["aviSegmentPath"]
+	utils.AviLog.Infof("key: %s, Adding LR %s, LS %s from networkinfo CR", objKey, lr, ls)
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("key: %s, Failed to get Cloud data from cache", objKey)
```

Comment:
is it just okay to return of the cloud object is not found or should there be some action? It seems like:
 - either we can fetch the cloud cache.
 - cloud itself does not exist in that case it's a `Fatalf`